pkg/asset/machines/master.go

```diff
@@ -181,7 +182,17 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 		mpool.Set(ic.Platform.OpenStack.DefaultMachinePlatform)
 		mpool.Set(pool.Platform.OpenStack)
 		pool.Platform.OpenStack = &mpool
-		machines, err = openstack.Machines(clusterID.InfraID, ic, pool, string(*rhcosImage), "master", "master-user-data")
+
+		imageName := clusterID.InfraID + "-rhcos"
+		// Here we check whether rhcosImage is a url or not. If this is the case, it means that the image was
+		// created by the installer and we have to use the universal name "<infraID>-rhcos". Otherwise, it means
+		// that we are given the name of the pre-created Glance image, which we should use for node provisioning.
+		_, err = url.ParseRequestURI(string(*rhcosImage))
+		if err != nil {
+			imageName = string(*rhcosImage)
+		}
```

Comment:
Should this be a function?

Reply:
I thought about it... Absolutely agree that it makes sense, but I don't know where to put this function properly. There is no a place for OpenStack util functions, unfortunately. I'll try to figure out where we can put it.